bpf/lbmap/lbmap.go

```diff
@@ -173,13 +208,94 @@ func LookupRevNat(key RevNatKey) (RevNatValue, error) {
 	return revnat.Convert(), nil
 }
 
+func gcd(x, y uint16) uint16 {
+	for y != 0 {
+		x, y = y, x%y
+	}
+	return x
+}
+
+type rrState struct {
+	index int
+	cw    uint16
+}
+
+func next(state *rrState, n int, max uint16, weights []uint16) uint16 {
+	for {
+		state.index = (state.index + 1) % n
+		if state.index == 0 {
+			state.cw = state.cw - 1
+			if state.cw <= 0 {
+				state.cw = max
+			}
+		}
+		if weights[state.index] >= state.cw {
```

Comment:
The datapath has a pre defined max sequence. So I can limit the loops to that count.